### DISCUSSION TITLE
Adblock filters not detected

### DISCUSSION THREAD

**Discussion by igorskyflyer** [github-linguist/linguist#6393] [Classification] [answered]
Hello, among other things, I have a repository where I store and maintain an Adblock filter list - actually a few of them. The repository is located at [AdVoid](https://github.com/igorskyflyer/ad-void) and as you can see the language bar only shows JavaScript as the language used in the repository.

One thing to note here is that all of my JavaScript files reside in the /docs folder, it's a React application and that folder is a rather new addition to the repository. A few months ago, I had only .txt filter lists and there was no language bar at all = no languages detected, thus, the React application is not the issue here. Also, what's weird as well is that the /docs folder contains HTML, CSS, JS and JSX files but only JS is detected.

Any insight is very much appreciated, thanks! 🙏🏼

**Comment by igorskyflyer** [comment, marked as answer]
Upon further research, I found a solution! 🙌🏼

If anyone has the same issue, all you have to do is to create a `.gitattributes` file in the root of your repo with the contents:

```shell
*.txt linguist-language=AdBlock linguist-detectable
```

The glob `*.txt` means all .txt files, if your filters are in a special directory in your repo, update the glob accordingly.

Take a look at my [.gitattributes](https://github.com/igorskyflyer/ad-void/blob/main/.gitattributes) file for more info.

**Reply by lildude** [reply]
This is expected as `.txt` files are considered `data` for all uses and in this case, you shouldn't need the language override (just the detectable override because `data` is ignored) as your files should match [this heuristic](https://github.com/github-linguist/linguist/blob/5a0c74277548122267d84283910abd5e0b89380e/lib/linguist/heuristics.yml#L728).

**Reply by igorskyflyer** [reply]
Thanks for the additional insight! That would work as well but I prefer to make it explicit.

**Reply by Rudxain** [reply]
https://github.com/github-linguist/linguist/blob/f164d13fa618023ecf2d8f2ed9a6ce5fae731346/docs/overrides.md#using-gitattributes
